File: parish360-core/src/main/java/org/parish360/core/dao/entities/common/GodParent.java
package org.parish360.core.dao.entities.common;

import jakarta.persistence.Embeddable;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Embeddable
public class GodParent {
    private String name;
    private String parish;
    private String baptismName;
    private String contact;
}

File: parish360_mobile/lib/core/common/entities/god_parent.dart
class GodParent {
  final String? name;
  final String? parish;
  final String? contact;
  final String? baptismName;

  GodParent({this.name, this.parish, this.contact, this.baptismName});

  factory GodParent.fromJson(Map<String, dynamic> json) {
    return GodParent(
      name: json['name'] as String?,
      parish: json['parish'] as String?,
      contact: json['contact'] as String?,
      baptismName: json['baptism_name'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'parish': parish,
      'contact': contact,
      'baptism_name': baptismName,
    };
  }
}
